test/cql-pytest/test_describe.py

```diff
@@ -552,6 +552,28 @@ def test_whitespaces_in_table_options(cql, test_keyspace, scylla_only):
         desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
         assert re.search(regex, desc) == None
 
+# In order to restore schema from `DESC SCHEMA`
+# UDTs have to be sorted topologically because they can depent on each other
+# The test doesn't check validity of UDT description, 
+# but only the order in `DESC KEYSPACE ks`(it's equivalent to part of `DESC SCHEMA`)
+def test_udts_order(scylla_only, cql, test_keyspace):
+    udt_names = [chr(ord('z') - i) + unique_name() for i in range(5)]
+    with new_type(cql, test_keyspace, "(a int)", udt_names[0]) as t0:
+        with new_type(cql, test_keyspace, f"(a frozen<{t0}>)", udt_names[1]) as t1:
+            with new_type(cql, test_keyspace, f"(a frozen<list<{t0}>>, b set<frozen<{t1}>>)", udt_names[2]) as t2:
+                with new_type(cql, test_keyspace, f"(a frozen<tuple<{t0}, {t2}>>)", udt_names[3]) as t3:
+                    with new_type(cql, test_keyspace, f"(a map<frozen<{t1}>, frozen<{t2}>>, b list<frozen<{t0}>>)", udt_names[4]) as t4:
+                        ordering_a = [t0, t1, t2, t3, t4]
+                        ordering_b = [t0, t1, t2, t4, t3]
+
+                        udt_order = []
+                        desc = cql.execute(f"DESC KEYSPACE {test_keyspace}")
+                        for row in desc:
+                            if row.name != test_keyspace:
+                                udt_order.append(f"{row.keyspace_name}.{row.name}")
+                        
+                        assert udt_order == ordering_a or udt_order == ordering_b                            
```

Comment:
If you will add additional test cases, this attempt to manually list all the correct orderings can quickly -become unwieldy. Instead, you can write a short piece code (that's the beauty of writing tests in Python) that verifies that for each type name, it's 1. listed in the definition, and 2. isn't referenced before it's defined. However, this checking code must be short and simple (one function) otherwise it too can be suspected to be buggy.